docs/design/proposals/adoption/resource-adoption.md

```diff
@@ -0,0 +1,201 @@
+
+# In Scope
+
+* Declaration of the new `AdoptedResource` CRD
+* Updates to generator for service-specific configuration
+* New reconciler for `AdoptedResource` CRs within each service
+* Validation webhook for `AdoptedResource` types
+* Unit and E2E tests with a specific ACK controller (e.g. S3 buckets)
+
+# Out of Scope
+
+* Integration of adopted resource types for every ACK controller
+* E2E tests for every ACK controller
+* Tooling for generating `AdoptedResource` CRs
+
+# Test Plan
+
+## Unit Tests
+
+* Assert validation webhook detects and rejects all blocked conditions
+* Assert adopted resource reconciler filters and manages associated CRs
+* Assert manager can create target CR type from description
+* Assert manager makes call to update AWS resource tags
+
+## E2E Tests
+
+Requirements:
+
+* 1 Kubernetes cluster
+* 1+ Existing AWS resources
+* 1 ACK controller (e.g. S3)
+    * Minimum 1 adoptable resource type
+    * Minimum 1 non-adoptable resource type
+
+Procedure:
+
+* Create `AdoptedResource` of target type that does not exist
+    * Assert validation webhook fails
+* Create `AdoptedResource` of target type that does exist
+    * Assert controller creates target type resource
+    * Assert created resource specification matches expected fields
+    * Assert AWS resource tags have been updated
+    * Assert status of created resource is stable
+* Create `AdoptedResource` that has the same metadata name as previous
+    * Assert validation webhook fails
+* Create `AdoptedResource` of type that explicitly blocks adoption
+    * Assert validation webhook fails
+
+# Discussions
+
+> How are users able to bulk import existing resources?
+
+For users that hope to bulk import existing resources, an automated way of creating `AdoptedResource` custom resources would expedite the initial effort. This automation is outside the scope of this implementation. However, given a user can create all of the resources, and installs all the necessary controllers, the method of adopting many resources fundamentally remains the same as importing them all singly.
+
+
+> How can I migrate across an ACK installation using adopted resources to maintain application state?
+
+The following workflow may apply to this transition:
+
+1. Export the identifiers (ARNs/IDs) from the current installation resources’ states
+2. Create `AdoptedResource` CRs for each of these identifiers
+3. Install and configure all necessary controllers to the new cluster
+4. Apply the `AdoptedResource` CRs to the cluster
+    1. Wait for all resources to end up in terminal state
+5. Delete the ACK controller deployments from the old installation
+    1. Do NOT delete the custom resources, this will delete the underlying AWS resources
+6. Apply existing manifests to override the newly created resources
+    1. *Note: If their state is already as desired, this should not make any updates to the AWS resources*
+7. Delete the custom resources from the old installation
```

Comment:
This step will necessitate manually removing the finalizers - through force deletion.